src/main/java/page_objects/BookTicketPage.java

```diff
@@ -0,0 +1,82 @@
+package page_objects;
+
+import common.Constants;
+import common.Ticket;
+import helpers.DriverHelper;
+import helpers.ElementHelper;
+import helpers.Wait;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.Select;
+
+public class BookTicketPage extends BasePage {
+    //Locators
+    private By drpDepartDate = By.cssSelector("select[name = 'Date']");
+    private By drpDepartFrom = By.cssSelector("select[name = 'DepartStation']");
+    private By drpArriveAt = By.cssSelector("select[name='ArriveStation']");
+    private By drpSeatType = By.cssSelector("select[name = 'SeatType']");
+    private By drpTicketAmount = By.cssSelector("select[name = 'TicketAmount']");
+    private By btnBookticket = By.cssSelector("input[value='Book ticket']");
+
+    //Elements
+    private Select getDrpDepartDate() {
+        return new Select(DriverHelper.getDriver().findElement(drpDepartDate));
+    }
+
+    private Select getDrpDepartFrom() {
+        return new Select(DriverHelper.getDriver().findElement(drpDepartFrom));
+    }
+
+    private Select getDrpArriveAt() {
+        return new Select(DriverHelper.getDriver().findElement(drpArriveAt));
+    }
+
+    private Select getDrpSeatType() {
+        return new Select(DriverHelper.getDriver().findElement(drpSeatType));
+    }
+
+    private Select getDrpTicketAmount() {
+        return new Select(DriverHelper.getDriver().findElement(drpTicketAmount));
+    }
+
+    private WebElement getBtnBookticket() {
+        return DriverHelper.getDriver().findElement(btnBookticket);
+    }
+
+    //Methods
+    public void selectDepartDate(String date) {
+        getDrpDepartDate().selectByVisibleText(date);
+    }
+
+    public void selectDepartFrom(String departStation) {
+        getDrpDepartFrom().selectByVisibleText(departStation);
+    }
+
+    public void selectArriveAt(String arriveStation) {
+        try {
+            getDrpArriveAt().selectByVisibleText(arriveStation);
+            Wait.untilElementIsSelected(DriverHelper.getDriver().findElement(drpArriveAt), Constants.TIME_WAIT);
+        } catch (org.openqa.selenium.StaleElementReferenceException ex) {
```

Comment:
why do we need to try catch?

Reply:
Because DOM changes if I select Depart from dropdown, so decided to use try catch it will attempt to find and select element at Arrive at dropdown

Reply:
I fixed it. Please help me check it. Thanks so much